packages/terminal/src/browser/terminal-quick-open-service.ts

```diff
@@ -0,0 +1,124 @@
+/********************************************************************************
+ * Copyright (C) 2019 Ericsson and others.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License v. 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0.
+ *
+ * This Source Code may also be made available under the following Secondary
+ * Licenses when the conditions for such availability set forth in the Eclipse
+ * Public License v. 2.0 are satisfied: GNU General Public License, version 2
+ * with the GNU Classpath Exception which is available at
+ * https://www.gnu.org/software/classpath/license.html.
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR GPL-2.0 WITH Classpath-exception-2.0
+ ********************************************************************************/
+
+import { inject, injectable } from 'inversify';
+import {
+    QuickOpenModel, QuickOpenItem, QuickOpenHandler,
+    QuickOpenOptions, QuickOpenItemOptions, QuickOpenMode,
+    PrefixQuickOpenService, ApplicationShell
+} from '@theia/core/lib/browser';
+import { WidgetManager } from '@theia/core/lib/browser';
+import { TerminalWidget } from './base/terminal-widget';
+import { TERMINAL_WIDGET_FACTORY_ID } from './terminal-widget-impl';
+
+@injectable()
+export class TerminalQuickOpenService implements QuickOpenModel, QuickOpenHandler {
+
+    @inject(PrefixQuickOpenService)
+    protected readonly prefixQuickOpenService: PrefixQuickOpenService;
+
+    @inject(WidgetManager)
+    protected readonly widgetManager: WidgetManager;
+
+    @inject(ApplicationShell)
+    protected readonly shell: ApplicationShell;
+
+    readonly prefix: string = 'term ';
+
+    get description(): string {
+        return 'Show All Opened Terminals';
+    }
+
+    getModel(): QuickOpenModel {
+        return this;
+    }
+
+    getOptions(): QuickOpenOptions {
+        return {
+            fuzzyMatchLabel: {
+                enableSeparateSubstringMatching: true
+            },
+            fuzzyMatchDescription: {
+                enableSeparateSubstringMatching: true
+            }
+        };
+    }
+
+    open(): void {
+        this.prefixQuickOpenService.open(this.prefix);
+    }
+
+    get all(): TerminalWidget[] {
+        return this.widgetManager.getWidgets(TERMINAL_WIDGET_FACTORY_ID) as TerminalWidget[];
+    }
+
+    async onType(lookFor: string, acceptor: (items: QuickOpenItem[]) => void): Promise<void> {
+        const terminalItems: QuickOpenItem[] = [];
+
+        const widgets: TerminalWidget[] = this.all.sort();
+
+        if (widgets.length === 0) {
```

Comment:
This item should not be displayed when there are no results.
It should be displayed at all times at the end of the list.